### DISCUSSION TITLE
[Help] paopaodns 只能用作局域网dns吗，家里宽带有公网用adh开启了doh给公司电脑用，但是公司电脑是电信宽带，查询的全是家里联通的记录，不能查询最近的服务器

### DISCUSSION THREAD

**Discussion by mrzj110** [kkkgo/PaoPaoDNS#9] [问答和帮助]
### 在提交之前，请确认

- [X] 我已经尝试搜索过Issue和看过参数说明，但没有找到相关问题。
- [X] 我正在使用最新的docker镜像版本（可以尝试`docker pull sliamb/paopaodns:latest`后重新创建容器，如果不确定的话可以尝试使用`sliamb/paopaodns:dev`）。

### 脚本自检日志

```txt
1
```

### 问题描述和复现步骤

paopaodns 只能用作局域网dns吗，家里宽带有公网用adh开启了doh给公司电脑用，但是公司电脑是电信宽带，查询的全是家里联通的记录，不能查询最近的服务器，本人在重庆。
![QQ截图20230427181440](https://user-images.githubusercontent.com/43603971/234832989-da1c01bb-77c1-4299-81c6-38e470bdd72e.png)

**Comment by kkkgo** [comment]
你可以阅读这篇文章了解递归DNS的工作原理：[《为啥需要递归DNS》](https://blog.03k.org/post/paopaodns.html)      
简单来说就是，递归DNS获取的DNS解析，是以你的出口IP和权威DNS沟通的结果，因此无论你用什么办法中转到你的DNS服务器，出口IP还是一样的，你DNS服务器用电信的宽带，就相当于电信运营商级别的DNS，你DNS服务器用联通宽带，就是联通的解析。

**Comment by mrzj110** [comment]
> 

mosdns有ecs功能，客户端查询携带的ecs信息查询，类似这个大佬的配置，https://github.com/pmkol/easymosdns

**Comment by kkkgo** [comment]
> > 
> 
> mosdns有ecs功能，客户端查询携带的ecs信息查询，类似这个大佬的配置，https://github.com/pmkol/easymosdns

上面的文章也有提到这一点。首先，ECS需要权威服务器支持，并不是所有权威服务器支持ECS。其次ECS是作为递归DNS和内网客户端之间的补偿选项，因为你不能和权威DNS直接沟通而折中的产物，如果你已经是递归DNS了，这个就没必要了。最后，就是考虑缓存储存的问题，如果这么弄的话首先要独立运行一套配置，然后redis也要单独跑一套，因为缓存的数据不一样，事情就很复杂了，而上面你提到的支持ECS的功能是因为他并不是递归DNS，数据不在自己手里，而是转发到公共递归DNS上的，就没这个问题，就如文章所述，因为公共DNS本身是有很多台不同线路的服务器的。

**Comment by mrzj110** [comment]
好的，谢谢解答
